### PR TITLE
Added custom icons in editor

### PR DESCRIPTION
**Pull request by nicolaisirghi**
### Changelog
New: Added custom icons

### CODE REVIEW

File: public/editor-client/src/api/index.ts [hunk 1]
```diff
@@ -1206,3 +1208,69 @@ export const updateGlobalBlocks = async (
...
+  });
+
+  if (response.ok) {
+    return response.json().then((r) => r.data);
+  }
+
+  throw new Error(t("Failed to upload icon"));
+};
+
+export const deleteIcon = async (uid: string): Promise<unknown> => {
```
Comment (victor-pg) [inline public/editor-client/src/api/index.ts:1257]: write concrete type instead of `unknown`

File: public/editor-client/src/config.ts [hunk 2]
```diff
@@ -158,6 +163,42 @@ const apiReader = parseStrict<PLUGIN_ENV["api"], API>({
...
+      Str.read
+    ),
+    throwOnNullish("Invalid API: uploadIconUrl")
+  ),
+  deleteIconUrl: pipe(
+    mPipe(
+      Obj.readKey("customIcon"),
+      Obj.read,
+      Obj.readKey("deleteIconUrl"),
+      Str.read
```
Comment (victor-pg) [inline public/editor-client/src/config.ts:199]: create an fn:
```ts
export const readIconUrl = (url:string)=>pipe(mPipe(Obj.readKey("customIcon", Obj.read, Obj.readKey(url), Str.read)) ,throwOnNullish(`Invalid API: ${url}`))
```
and here use simple:
```ts
deleteIconUrl: readIconUrl("deleteIconUrl")
```

**Review by victor-pg — CHANGES_REQUESTED** [review]

File: public/editor-client/src/customIcon/index.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      });
+    } catch (e) {
+      rej(t("Icon upload cancelled"));
+    }
+  },
+
+  async delete(res, rej, { uid }) {
+    apiDeleteIcon(uid)
+      .then(() => res(uid))
+      .catch(() => rej(t("Failed to delete icon")));
```
Comment (maxval1) [inline public/editor-client/src/customIcon/index.ts:51]: 1. async/await
2. try/catch

File: public/editor-client/src/api/index.ts [hunk 4]
```diff
@@ -1206,3 +1208,69 @@ export const updateGlobalBlocks = async (
...
+  const url = makeUrl(api.iconsUrl, {
+    "orderBy[id]": "DESC",
+    count: "1000"
+  });
+
+  return request(url, {
+    method: "GET"
+  })
+    .then((r) => r.json())
+    .then((r) => r.data);
```
Comment (maxval1) [inline public/editor-client/src/api/index.ts:1229]: 1. async/await
2. try/catch

File: public/editor-client/src/index.ts [hunk 5]
```diff
@@ -47,6 +48,10 @@ const api = {
     addFile,
     fileUrl: config.api.fileUrl
   },
+  customIcon: {
+    ...customIcon,
+    getIconUrl: config.api.getIconUrl
```
Comment (maxval1) [inline public/editor-client/src/index.ts:53]: rename to `iconUrl`

**Review by maxval1 — CHANGES_REQUESTED** [review]

File: public/editor-client/src/index.ts [hunk 6]
```diff
@@ -1,3 +1,4 @@
+import { merge } from "lodash";
```
Comment (maxval1) [inline public/editor-client/src/index.ts:1]: same as `lodash/set`

**Review by maxval1 — CHANGES_REQUESTED** [review]

**Pull request merged**